scripts/make_spec.js

```diff
@@ -1,211 +1,210 @@
-        stakingEpochDuration, // _stakingEpochDuration
-        collectRoundLength    // _collectRoundLength
-    ]});
-    spec.accounts['0x7000000000000000000000000000000000000000'] = {
-        balance: '0',
-        constructor: await deploy.encodeABI()
-    };
-
-    console.log('Saving spec.json file ...');
-    fs.writeFileSync('spec.json', JSON.stringify(spec, null, '  '));
-    console.log('Done');
-}
-
-async function compile(dir, contractName) {
-    const input = {
-        language: 'Solidity',
-        sources: {
-            '': {
-                content: fs.readFileSync(dir + contractName + '.sol').toString()
-            }
-        },
-        settings: {
-            optimizer: {
-                enabled: true,
-                runs: 200
-            },
-            evmVersion: "constantinople",
-            outputSelection: {
-                '*': {
-                    '*': [ 'abi', 'evm.bytecode.object' ]
-                }
-            }
-        }
-    }
-
-    const compiled = JSON.parse(solc.compile(JSON.stringify(input), function(path) {
-        let content;
-        try {
-            content = fs.readFileSync(dir + path);
-        } catch (e) {
-            if (e.code == 'ENOENT') {
-                content = fs.readFileSync(dir + '../' + path);
-            }
-        }
-        return {
-            contents: content.toString()
-        }
-    }));
-
-    const result = compiled.contracts[''][contractName];
-
-    return {abi: result.abi, bytecode: result.evm.bytecode.object};
-}
-
-// NETWORK_NAME=DPoSChain NETWORK_ID=101 OWNER=0x1092a1E3A3F2FB2024830Dd12064a4B33fF8EbAe INITIAL_VALIDATORS=0xeE385a1df869A468883107B0C06fA8791b28A04f,0x71385ae87c4b93db96f02f952be1f7a63f6057a6,0x190ec582090ae24284989af812f6b2c93f768ecd STAKING_EPOCH_DURATION=120960 COLLECT_ROUND_LENGTH=200 node scripts/make_spec.js
+const fs = require('fs');
+const path = require('path');
+const solc = require('solc');
+const Web3 = require('web3');
+const web3 = new Web3(new Web3.providers.HttpProvider("https://a.invalid"));
+
+main();
+
+async function main() {
+    const networkName = process.env.NETWORK_NAME;
+    const networkID = process.env.NETWORK_ID;
+    const owner = process.env.OWNER.trim();
+    const initialValidators = process.env.INITIAL_VALIDATORS.split(',');
+    for (let i = 0; i < initialValidators.length; i++) {
+        initialValidators[i] = initialValidators[i].trim();
+    }
+    const stakingEpochDuration = process.env.STAKING_EPOCH_DURATION;
+    const collectRoundLength = process.env.COLLECT_ROUND_LENGTH;
+
+    const contracts = [
+        'EternalStorageProxy',
+        'ValidatorSetAuRa',
+        'BlockRewardAuRa',
+        'RandomAuRa',
+        'TxPermission',
+        'Certifier',
+        'Registry',
+        'InitializerAuRa'
+    ];
+
+    console.log(`Loading spec.json from poa-chain-spec repo...`);
+    const spec = JSON.parse(fs.readFileSync(path.join(__dirname, 'template.json'), 'UTF-8'));
```

Comment:
Let's create `templates` directory, move `template.json` there, and rename `template.json` to `spec.json`.